docs/pages/pmd/userdocs/extending/rule_guidelines.md

```diff
@@ -12,26 +12,37 @@ Here is a bunch of thing to do you may consider once your rule is “up and runn
 
 ## How to define rules priority
 
-Rule priority may, of course, changes a lot depending on the context of the project. However, you can use the following guidelines to assert the legitimate priority of your rule:
+Rule priority may, of course, changes a lot depending on the context of the project. However, you can use the
+following guidelines to assert the legitimate priority of your rule:
 
-1.  **Change absolutely required.** Behavior is critically broken/buggy.
-2.  **Change highly recommended.** Behavior is quite likely to be broken/buggy.
-3.  **Change recommended.** Behavior is confusing, perhaps buggy, and/or against standards/best practices.
-4.  **Change optional.** Behavior is not likely to be buggy, but more just flies in the face of standards/style/good taste.
-5.  **Change highly optional.** Nice to have, such as a consistent naming policy for package/class/fields…
+1.  **High: Change absolutely required.** Behavior is critically broken/buggy.
+2.  **Medium High: Change highly recommended.** Behavior is quite likely to be broken/buggy.
+3.  **Medium: Change recommended.** Behavior is confusing, perhaps buggy, and/or against standards/best practices.
+4.  **Medium Low: Change optional.** Behavior is not likely to be buggy, but more just flies in the face of
+    standards/style/good taste.
+5.  **Low: Change highly optional.** Nice to have, such as a consistent naming policy for package/class/fields…
```

Comment:
notice almost all our rules are priority 3 (Medium) by default, we don't really follow this ourselves

Reply:
While that's true, in my projects, I like to assign my own priority value for each rule so that my developers know what to focus on first... and then I can put a `minimum-priority` flag in the CLI and ignore lower-priority items.

Reply:
related issues: #4327, #3926